jmespath-core/src/main/java/io/burt/jmespath/jcf/JcfAdapter.java

```diff
@@ -143,6 +154,15 @@ public Object getProperty(Object value, String name) {
   }
 
   @Override
+  public Collection<String> getPropertyNames(Object value) {
+    if (isObject(value)) {
+      return ((Map<String, Object>) value).keySet();
```

Comment:
And come to think of it, `JsonParser` uses `#createObject`, so in all cases the argument to this method should have been created by `#createObject`. If it wasn't then someone is using this class for things it wasn't intended to be used for.